ExpSwift/Classes/Exp.swift

```diff
@@ -1057,7 +1149,20 @@ private func refreshAuthToken(_ result:Auth){
             refreshAuthToken(result)
         }
     }
+}
 
+/**
+ Refresh Auth Token Recursive with Timeout
+ */
+private func refreshAuthToken(_ expirationToken:String){
+    after(interval: getTimeout(Double(expirationToken)!)).then{ result -> Void in
```

Comment:
there should probably be a default here for getTimeout() or the expirationToken assuming it fails you don't want this going into an infinite loop with timeout 0 or something.